dense_vector/track.py

```diff
@@ -1,5 +1,110 @@
+import functools
 import json
+import logging
 import os
+import re
+from collections import defaultdict
+from typing import Dict, List
+
+from esrally.track import loader
+from esrally.track.track import Parallel, Task
+
+logger = logging.getLogger(__name__)
+
+
+def load_query_vectors(queries_file):
+    if not (os.path.exists(queries_file) and os.path.isfile(queries_file)):
+        raise ValueError(f"Provided queries file '{queries_file}' does not exist or is not a file")
+    query_vectors: Dict[str, List[float]]
+    with open(queries_file, "r") as f:
+        logger.debug(f"Reading query vectors from '{queries_file}'")
+        lines = f.readlines()
+        query_vectors = {_index: json.loads(vector) for _index, vector in enumerate(lines)}
+        logger.debug(f"Finished reading query vectors from '{queries_file}'")
+    return query_vectors
+
+
+async def extract_exact_neighbors(query_vector: List[float], index: str, max_size: int, client) -> List[str]:
+    script_query = await client.search(
+        body={
+            "query": {
+                "script_score": {
+                    "query": {"match_all": {}},
+                    "script": {
+                        "source": "cosineSimilarity(params.query, 'vector') + 1.0",
```

Comment:
This method (and the whole `KnnVectorStore`) could be a bit more generic by passing the vector field as a parameter as well, and reuse it directly in other tracks as well (e.g. `so_vector`).